docs/compass/03-packages-credential-requests.md

```diff
@@ -0,0 +1,137 @@
+# Credential requests for Packages
+
+## Overview
+
+On Runtime, Application is represented as Service Class, and every Package within Application is represented as Service Plan. This document describes credential requests for APIs during the Service Plan provisioning. It also mentions passing optional input parameters from Kyma Runtime to Application or Integration System during the provisioning process.
+
+## Assumptions
+
+- Multiple Service Instances can be created from a given Package within Application.
+- During Service Class provisioning, the provided input has to be sent back to Integration System or Application via Compass. The reason is that there is no trusted connection between Integration System and Runtime.
+- Passing input parameters is done during requesting credentials for a given Service Instance.
+
+## Details
+
+The Director GraphQL API is updated to store credentials per Service Instance. Credentials for every Instance across all Runtimes are stored on the Package level.
+
+### API Credentials Flow
+
+This diagram illustrates the API credentials flow in details. The Application provides Webhook API where Management Plane requests for providing new credentials for given Package.
+
+> **NOTE:** There is an option that Application does not support Webhook API. That means Application needs to monitor registered API Definitions and set API credentials when new Runtime assigned. The Administrator can exchange credentials for registered APIs at any time too.
+
+![Application Webhook](./assets/api-credentials-flow.svg)
+
+Assume we have Application which is already registered into Management Plane. No Runtimes are assigned yet. Application has one Package which contains single API Definition.
+
+1. The Administrator requests new Runtime with Application via Cockpit.
```

Comment:
why this step is described here? how provisioning runtime is connected with PAckages? Why we mention Cockpit here, is it important at all?

Reply:
I rewrote this section